# 파이썬 데이터분석 프로젝트

## 데이터 딕셔너리

Season	Match Season

DateTime	Match Date and Time (yyyy-mm-dd hh:mm:ss)

HomeTeam	Home Team

AwayTeam	Away Team

FTHG	Full Time Home Team Goals

FTAG	Full Time Away Team Goals

FTR	Full Time Result (H=Home Win, D=Draw, A=Away Win)

HTHG	Half Time Home Team Goals

HTAG	Half Time Away Team Goals

HTR	Half Time Result (H=Home Win, D=Draw, A=Away Win)

Referee	Match Referee

HS	Home Team Shots

AS	Away Team Shots

HST	Home Team Shots on Target

AST	Away Team Shots on Target

HC	Home Team Corners

AC	Away Team Corners

HF	Home Team Fouls Committed

AF	Away Team Fouls Committed

HY	Home Team Yellow Cards

AY	Away Team Yellow Cards

HR	Home Team Red Cards

AR	Away Team Red Cards

In [224]:
import math
import pandas as pd
import numpy as np

In [2]:
leage = pd.read_csv('archive/results.csv', encoding='cp949')

In [3]:
#데이터 확인
leage.tail()

Season              DateTime     HomeTeam        AwayTeam  FTHG  FTAG   
11108  2021-22  2022-04-09T17:30:00Z  Aston Villa       Tottenham     0     4  \
11109  2021-22  2022-04-10T14:00:00Z    Brentford        West Ham     2     0   
11110  2021-22  2022-04-10T14:00:00Z    Leicester  Crystal Palace     2     1   
11111  2021-22  2022-04-10T14:00:00Z      Norwich         Burnley     2     0   
11112  2021-22  2022-04-10T16:30:00Z     Man City       Liverpool     2     2   

      FTR  HTHG  HTAG HTR  ...  HST  AST   HC   AC    HF    AF   HY   AY   HR   
11108   A   0.0   1.0   A  ...  8.0  5.0  9.0  3.0  12.0  14.0  2.0  3.0  0.0  \
11109   H   0.0   0.0   D  ...  7.0  1.0  4.0  6.0   2.0   6.0  0.0  1.0  0.0   
11110   H   2.0   0.0   H  ...  3.0  3.0  3.0  4.0  11.0  12.0  1.0  1.0  0.0   
11111   H   1.0   0.0   H  ...  6.0  4.0  6.0  7.0  12.0  10.0  1.0  1.0  0.0   
11112   D   2.0   1.0   H  ...  5.0  4.0  4.0  1.0   9.0  11.0  1.0  4.0  0.0   

        AR  
11108  0.0  
11109  0.0  
11110  0.0  
11111  0.0  
11112  0.0  

[5 rows x 23 columns]

In [4]:
leage.HomeTeam.value_counts()

HomeTeam
Arsenal             552
Man United          552
Liverpool           551
Everton             551
Tottenham           551
Chelsea             550
Newcastle           513
West Ham            495
Aston Villa         494
Man City            456
Southampton         419
Blackburn           327
Sunderland          304
Leicester           302
Fulham              285
Middlesbrough       266
Leeds               248
West Brom           247
Bolton              247
Crystal Palace      227
Stoke               190
Norwich             172
Coventry            156
Wigan               152
Charlton            152
Wolves              149
Burnley             148
Watford             148
Sheffield Weds      137
Wimbledon           137
Birmingham          133
Swansea             133
Derby               133
Portsmouth          133
QPR                 118
Hull                 95
Bournemouth          95
Brighton             92
Ipswich              80
Sheffield United     78
Nott'm Forest        78
Reading

In [5]:
leage.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11113 entries, 0 to 11112
Data columns (total 23 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Season    11113 non-null  object 
 1   DateTime  11113 non-null  object 
 2   HomeTeam  11113 non-null  object 
 3   AwayTeam  11113 non-null  object 
 4   FTHG      11113 non-null  int64  
 5   FTAG      11113 non-null  int64  
 6   FTR       11113 non-null  object 
 7   HTHG      10189 non-null  float64
 8   HTAG      10189 non-null  float64
 9   HTR       10189 non-null  object 
 10  Referee   8289 non-null   object 
 11  HS        8289 non-null   float64
 12  AS        8289 non-null   float64
 13  HST       8289 non-null   float64
 14  AST       8289 non-null   float64
 15  HC        8289 non-null   float64
 16  AC        8289 non-null   float64
 17  HF        8289 non-null   float64
 18  AF        8289 non-null   float64
 19  HY        8289 non-null   float64
 20  AY        8289 non-null   fl

In [6]:
leage.sort_values(by = ['Season', 'DateTime'])
#너무 과거의 시즌은 제외하도록 합니다
leage_00 = leage[leage.Season >= '2000-01']

## 토트넘의 경기 결과를 도출하도록 합니다

In [7]:
leage_tot = leage_00[(leage_00.HomeTeam == 'Tottenham') | (leage_00.AwayTeam == 'Tottenham')]
leage_tot_home = leage_tot[leage_tot.HomeTeam == 'Tottenham']
leage_tot_away = leage_tot[leage_tot.AwayTeam == 'Tottenham']

### 00-01 시즌, 09-10 시즌, 20-21시즌의 우승팀 데이터를 추출합니다

In [8]:
#00-01 시즌 우승팀: 맨체스터 유나이티드
#09-10 시즌 우승팀: 첼시
#20-21 시즌 우승팀: 맨체스터 시티

leage_winner_00_01_home = leage_00[(leage_00.HomeTeam == 'Man United') & (leage_00.Season == '2000-01')]
leage_winner_09_10_home = leage_00[(leage_00.HomeTeam == 'Chelsea') & (leage_00.Season == '2009-10')]
leage_winner_20_21_home = leage_00[(leage_00.HomeTeam == 'Man City') & (leage_00.Season == '2020-21')]
leage_winner_00_01_away = leage_00[(leage_00.AwayTeam == 'Man United') & (leage_00.Season == '2000-01')]
leage_winner_09_10_away = leage_00[(leage_00.AwayTeam == 'Chelsea') & (leage_00.Season == '2009-10')]
leage_winner_20_21_away = leage_00[(leage_00.AwayTeam == 'Man City') & (leage_00.Season == '2020-21')]

### 각 3시즌의 우승팀들의 공격력과 수비력을 추출합니다

In [148]:
print(
    '홈: 득점과 유효슈팅\n',
    leage_winner_00_01_home.loc[:,['FTHG', 'HST']].mean(), '\n',
    leage_winner_09_10_home.loc[:,['FTHG', 'HST']].mean(), '\n',
    leage_winner_20_21_home.loc[:,['FTHG', 'HST']].mean(), '\n',
    sep = ''
)
print(
    '홈: 실점과 상대 유효슈팅\n',
    leage_winner_00_01_home.loc[:,['FTAG', 'AST']].mean(), '\n',
    leage_winner_09_10_home.loc[:,['FTAG', 'AST']].mean(), '\n',
    leage_winner_20_21_home.loc[:,['FTAG', 'AST']].mean(), '\n',
    sep = ''
)
print(
    '원정: 득점과 유효슈팅\n',
    leage_winner_00_01_away.loc[:,['FTAG', 'AST']].mean(), '\n',
    leage_winner_09_10_away.loc[:,['FTAG', 'AST']].mean(), '\n',
    leage_winner_20_21_away.loc[:,['FTAG', 'AST']].mean(), '\n',
    sep = ''
)
print(
    '원정: 실점과 상대 유효슈팅\n',
    leage_winner_00_01_away.loc[:,['FTHG', 'HST']].mean(), '\n',
    leage_winner_09_10_away.loc[:,['FTHG', 'HST']].mean(), '\n',
    leage_winner_20_21_away.loc[:,['FTHG', 'HST']].mean(), '\n',
    sep = ''
)

#프리미어리그 평균도 구해 봅시다
print(
    '프리미어리그 평균',
    leage_00.loc[:,['FTHG', 'HST']].mean(), 
    leage_00.loc[:,['FTAG', 'AST']].mean(), 
    sep = '\n'
)

홈: 득점과 유효슈팅
FTHG    2.578947
HST     8.000000
dtype: float64
FTHG     3.578947
HST     12.105263
dtype: float64
FTHG    2.263158
HST     6.315789
dtype: float64

홈: 실점과 상대 유효슈팅
FTAG    0.631579
AST     3.157895
dtype: float64
FTAG    0.736842
AST     3.684211
dtype: float64
FTAG    0.894737
AST     2.789474
dtype: float64

원정: 득점과 유효슈팅
FTAG    1.578947
AST     5.894737
dtype: float64
FTAG    1.842105
AST     8.315789
dtype: float64
FTAG    2.105263
AST     5.210526
dtype: float64

원정: 실점과 상대 유효슈팅
FTHG    1.000000
HST     4.421053
dtype: float64
FTHG    0.947368
HST     4.684211
dtype: float64
FTHG    0.789474
HST     2.473684
dtype: float64

프리미어리그 평균
FTHG    1.519242
HST     6.117264
dtype: float64
FTAG    1.155507
AST     4.768247
dtype: float64


#### 결론적으로 우승팀의 기록을 살펴보면

홈 경기에서 평균 2.7의 득점을, 약 9번의 유효슈팅을 했다. 약 0.7점을 실점했고 약 3번의 유효슈팅을 허용했다.

원정 경기에서 약 1.8점을 득점했고 약 6.5번의 유효슈팅을 했다. 약 0.9점을 실점했고 약 4번의 유효슈팅을 허용했다.

In [68]:
#토트넘의 평균적인 성적도 추출해 봅시다.

print(
    '홈 경기: 토트넘의 득점과 유효슈팅\n',
    leage_tot_home.loc[:,['FTHG', 'HST']].mean(), '\n',
    '홈 경기: 토트넘의 실점과 상대 유효슈팅\n',
    leage_tot_home.loc[:,['FTAG', 'AST']].mean(), '\n',
    '원정 경기: 토트넘의 득점과 유효슈팅\n',
    leage_tot_away.loc[:,['FTAG', 'AST']].mean(), '\n',
    '원정 경기: 토트넘의 실점과 상대 유효슈팅\n',
    leage_tot_away.loc[:,['FTHG', 'HST']].mean(), '\n',
    sep = ''
)

홈 경기: 토트넘의 득점과 유효슈팅
FTHG    1.814010
HST     7.328502
dtype: float64
홈 경기: 토트넘의 실점과 상대 유효슈팅
FTAG    1.021739
AST     4.688406
dtype: float64
원정 경기: 토트넘의 득점과 유효슈팅
FTAG    1.346988
AST     5.431325
dtype: float64
원정 경기: 토트넘의 실점과 상대 유효슈팅
FTHG    1.409639
HST     5.877108
dtype: float64



#### 토트넘의 기록을 우승팀의 기록과 비교하면 

홈 경기에서 평균 1.8(우승팀 2.7)의 득점을, 약 7.3(우승팀 9)번의 유효슈팅을 했다. 약 1(우승팀 0.7)점을 실점했고 약 4.6(우승팀 3)번의 유효슈팅을 허용했다.

원정 경기에서 약 1.35(우승팀 1.8)점을 득점했고 약 5.4(우승팀 6.5)번의 유효슈팅을 했다. 약 1.4(우승팀 0.9)점을 실점했고 약 5.8(우승팀 4)번의 유효슈팅을 허용했다.

### 즉

토트넘은 우승팀의 공격력과 수비력에 비해 상당히 낮은 수치를 보여 주었다.

# 이번엔 공격력 부분을 살펴보도록 합니다.

슈팅 시도 횟수 대비 유효수팅을 몇 개 만들어 내는지를 비교해 봅시다.

In [230]:
winner_shot_hit = [
    (leage_winner_00_01_home.loc[:, 'HST'] / leage_winner_00_01_home.loc[:, 'HS']).mean(),
    (leage_winner_09_10_home.loc[:, 'HST'] / leage_winner_09_10_home.loc[:, 'HS']).mean(),
    (leage_winner_20_21_home.loc[:, 'HST'] / leage_winner_20_21_home.loc[:, 'HS']).mean(),
    (leage_winner_00_01_away.loc[:, 'AST'] / leage_winner_00_01_away.loc[:, 'AS']).mean(),
    (leage_winner_09_10_away.loc[:, 'AST'] / leage_winner_09_10_away.loc[:, 'AS']).mean(),
    (leage_winner_20_21_away.loc[:, 'AST'] / leage_winner_20_21_away.loc[:, 'AS']).mean(),
]
print(f'세 시즌 우승팀의 평균 슈팅 대비 유효슈팅 전환 비율: {np.mean(winner_shot_hit) * 100:.3f}%')

tot_shot_hit = [
    (leage_tot_home.loc[:, 'HST'] / leage_tot_home.loc[:, 'HS']).mean(),
    (leage_tot_away.loc[:, 'AST'] / leage_tot_away.loc[:, 'AS']).mean(),
]
print(f'토트넘의 평균 슈팅 대비 유효슈팅 전환 비율: {np.mean(tot_shot_hit) * 100:.3f}%')

leage_shot_hit = [
    (leage_00.loc[:, 'HST'] / leage_00.loc[:, 'HS']).replace(np.inf, 0).mean(),
    (leage_00.loc[:, 'AST'] / leage_00.loc[:, 'AS']).replace(np.inf, 0).mean(),
]
print(f'프리미어리그 팀의 평균 슈팅 대비 유효슈팅 전환 비율: {np.mean(leage_shot_hit) * 100:.3f}%')


세 시즌 우승팀의 평균 슈팅 대비 유효슈팅 전환 비율: 47.722%
토트넘의 평균 슈팅 대비 유효슈팅 전환 비율: 47.955%
프리미어리그 팀의 평균 슈팅 대비 유효슈팅 전환 비율: 45.320%


### 슈팅 대비 유효슈팅 전환 비율은 토트넘의 평균과 우승팀의 평균이 그렇게 차이나지 않았습니다.

이것은 토트넘 공격진의 결정력은 우승팀들과 견주어 볼 때 크게 떨어지지 않았다는 것을 말합니다.

실제로도 토트넘은 준수한 공격진을 항상 배출해 낸 것으로 유명합니다.

그렇다면 이번에는 슈팅 수 자체가 얼마나 차이나는지 비교해 보겠습니다.

In [238]:
winner_shot = [
    leage_winner_00_01_home.loc[:, 'HS'].mean(),
    leage_winner_09_10_home.loc[:, 'HS'].mean(),
    leage_winner_20_21_home.loc[:, 'HS'].mean(),
    leage_winner_00_01_away.loc[:, 'AS'].mean(),
    leage_winner_09_10_away.loc[:, 'AS'].mean(),
    leage_winner_20_21_away.loc[:, 'AS'].mean()
]
print(f'우승팀들의 경기 당 평균 슈팅은 {np.mean(winner_shot):.2f}번 이었습니다.')

tot_shot = [
    leage_tot_home.loc[:, 'HS'].mean(),
    leage_tot_away.loc[:, 'AS'].mean()
]
print(f'그에 비해 토트넘의 경기 당 평균 슈팅은 {np.mean(tot_shot):.2f}번 이었습니다.')

leage_00_shot = [
    leage_00.loc[:, 'HS'].mean(),
    leage_00.loc[:, 'AS'].mean()
]
print(f'프리미어리그 팀의 평균 슈팅 횟수는 {np.mean(leage_00_shot):.2f}번 이었습니다.')

우승팀들의 경기 당 평균 슈팅은 16.29번 이었습니다.
그에 비해 토트넘의 경기 당 평균 슈팅은 13.50번 이었습니다.
프리미어리그 팀의 평균 슈팅 횟수는 12.07번 이었습니다.


경기 당 슈팅을 시도하는 횟수 자체가 평균적으로 3번이 적은 것을 알 수 있었습니다.

이는 토트넘의 전체적인 선수단의 기회 창출 및 슈팅 기회 창출 능력이 우승팀들에 비해 떨어진다는 것을 나타냅니다.

이번에는 슈팅 횟수가 8회보다 적었던 경기의 수를 비교해 봅니다. 평균치가 비슷하더라도 기회를 거의 만들어내지 못한 경기의 수가 많다면 그 경기에서 패배하거나 비길 확률이 더 높을 것이라고 유추해 볼 수 있습니다.

In [246]:
winner_nojam_game = [
    len(leage_winner_00_01_home[leage_winner_00_01_home.HS < 8].HS),
    len(leage_winner_00_01_away[leage_winner_00_01_away.AS < 8].AS),
    len(leage_winner_09_10_home[leage_winner_09_10_home.HS < 8].HS),
    len(leage_winner_09_10_away[leage_winner_09_10_away.AS < 8].AS),
    len(leage_winner_20_21_home[leage_winner_20_21_home.HS < 8].HS),
    len(leage_winner_20_21_away[leage_winner_20_21_away.AS < 8].AS),
]

print(f'우승팀은 평균적으로 시즌당 {np.mean(winner_nojam_game):.3f}번의 공격력이 저조한 경기가 있었습니다.')

tot_nojam_game = [
    len(leage_tot_home[leage_tot_home.HS < 8].HS),
    len(leage_tot_away[leage_tot_away.AS < 8].AS)
]

print(f'토트넘은 평균적으로 시즌당 {sum(tot_nojam_game) / (829 / 38):.3f}번의 공격력이 저조한 경기가 있었습니다')

leage_00_nojam_game = (len(leage_00[(leage_00.HS < 8) | (leage_00.AS < 8)])) / (len(leage_00) * 2 / 38)

print(f'프리미어리어리그 팀은 평균적으로 시즌당 {leage_00_nojam_game:.3f}번의 공격력이 저조한 경기가 있었습니다.')

우승팀은 평균적으로 시즌당 0.833번의 공격력이 저조한 경기가 있었습니다.
토트넘은 평균적으로 시즌당 4.951번의 공격력이 저조한 경기가 있었습니다
프리미어리어리그 팀은 평균적으로 시즌당 6.922번의 공격력이 저조한 경기가 있었습니다.


### 드디어 확실히 의미가 있는 차이점을 찾아내었습니다.

유효슈팅이 아닌 일반 슈팅 시도가 8번 이하인 경기의 수가 5배 이상 차이나는 것을 알 수 있습니다.

따라서 토트넘은 시즌 중 기복이 심한 경기력을 보이거나 강팀을 상대로 일정 수준 이상의 공격도 하지 못하고 일방적으로 패배하는 경기가 시즌 중 평균적으로 5회 정도 나타난다고 해석할 수 있습니다. 이 5번의 경기를 모두 패배하거나 비겼다고 가정할 때, 이러한 특징으로 인해 최대 15점에서 최소 10점의 승점을 잃었다는 것을 가정할 수 있습니다.

실제로 프리미어리그에서 슈팅 수가 8개보다 낮을 때, 승률을 분석해 보겠습니다.

In [129]:
nojam_game_winrate = (
    len(leage_00[((leage_00.HS < 8) & (leage_00.FTR == 'H')) | ((leage_00.AS < 8) & (leage_00.FTR == 'A'))])
    /
    len(leage_00[(leage_00.HS < 8) | (leage_00.AS < 8)])
)
print(f'슈팅 수가 8개 이하일 때 프리미어리그팀이 승리를 거둔 경기는 {nojam_game_winrate * 100:.1f}%였습니다.')

슈팅 수가 8개 이하일 때 프리미어리그팀이 승리를 거둔 경기는 19.4%였습니다.


## 토트넘의 공격에 관한 분석을 정리해 보겠습니다.

| \ | 우승팀 분석 | 토트넘 분석 | 프리미어리그 평균 |
|:------------------------|:---------------:|:---------------:|:---------------:|
|홈 득점|2.7|1.8|1.5|
|원정 득점|1.8|1.35|1.15|
|홈 유효 슈팅|9|7.3|6.1|
|원정 유효 슈팅|6.5|5.4|4.7|
|슈팅 대비 유효슈팅 비율|47.722%|47.955%|45.320%|
|평균 슈팅 횟수|16.29|13.50|12.07|
|저조한 경기력|0.833|<span style = "color : red">4.951</span>|6.922|


# 이번엔 토트넘의 수비 데이터를 분석해 보겠습니다.

먼저 실점 자체를 비교해 보겠습니다.

In [144]:
#위에서 사용한 코드를 재사용합시다.
# print(
#     '홈: 실점과 상대 유효슈팅\n',
#     leage_winner_00_01_home.loc[:,['FTAG', 'AST']].mean(), '\n',
#     leage_winner_09_10_home.loc[:,['FTAG', 'AST']].mean(), '\n',
#     leage_winner_20_21_home.loc[:,['FTAG', 'AST']].mean(), '\n',
#     sep = ''
# )
# print(
#     '원정: 실점과 상대 유효슈팅\n',
#     leage_winner_00_01_away.loc[:,['FTHG', 'HST']].mean(), '\n',
#     leage_winner_09_10_away.loc[:,['FTHG', 'HST']].mean(), '\n',
#     leage_winner_20_21_away.loc[:,['FTHG', 'HST']].mean(), '\n',
#     sep = ''
# )

winner_lose_score = [
    leage_winner_00_01_home.loc[:,['FTAG']].mean(),
    leage_winner_00_01_away.loc[:,['FTHG']].mean(),
    leage_winner_09_10_home.loc[:,['FTAG']].mean(),
    leage_winner_09_10_away.loc[:,['FTHG']].mean(),
    leage_winner_20_21_home.loc[:,['FTAG']].mean(),
    leage_winner_20_21_away.loc[:,['FTHG']].mean(),
]
print(f'우승팀들의 경기당 평균 실점은 {np.mean(winner_lose_score):.3f}점이었습니다.')

tot_lose_score = [
    leage_tot_home.FTAG.mean(),
    leage_tot_away.FTHG.mean()
]
print(f'토트넘의 경기당 평균 실점은 {np.mean(tot_lose_score):.3f}점이었습니다.')

print(f'프리미어리그 팀의 평균 실점은 {leage_00.FTAG.mean():.3f}점이었습니다.')

우승팀들의 경기당 평균 실점은 0.833점이었습니다.
토트넘의 경기당 평균 실점은 1.216점이었습니다.
프리미어리그 팀의 평균 실점은 1.156점이었습니다.


### 토트넘은 프리미어리그 평균 실점보다 더 많은 점수를 실점하고 있었습니다.

벌써부터 수비력에 문제가 있다는 예상을 할 수 있습니다.

수비력을 조금 더 자세히 살펴보며 어떤 약점이 있는지 더 알아보겠습니다.

이번에는 경기당 상대 슈팅 횟수와 실점 간의 관계에 대해 살펴보겠습니다.

In [271]:
#먼저 상대팀에게 몇 번의 슈팅을 허용하는지를 보겠습니다.

winner_enermy_shot = [
    leage_winner_00_01_home.loc[:,['AS']],
    leage_winner_00_01_away.loc[:,['HS']],
    leage_winner_09_10_home.loc[:,['AS']],
    leage_winner_09_10_away.loc[:,['HS']],
    leage_winner_20_21_home.loc[:,['AS']],
    leage_winner_20_21_away.loc[:,['HS']],
]

print(f'우승팀은 경기당 상대에게 평균 {np.mean(winner_enermy_shot):.2f}번의 슈팅을 허용했습니다.')

tot_enermy_shot = [
    leage_tot_home.loc[:,['AS']].mean(),
    leage_tot_away.loc[:,['HS']].mean()
]

print(f'토트넘은 경기당 상대에게 평균 {np.mean(tot_enermy_shot):.2f}번의 슈팅을 허용했습니다.')

leage_00_enermy_shot = [
    leage_00.loc[:, 'HS'].mean(),
    leage_00.loc[:, 'AS'].mean()
]
print(f'프리미어리그 팀은 상대에게 평균 {np.mean(leage_00_enermy_shot):.2f}번의 슈팅을 허용했습니다.')

우승팀은 경기당 상대에게 평균 8.14번의 슈팅을 허용했습니다.
토트넘은 경기당 상대에게 평균 11.51번의 슈팅을 허용했습니다.
프리미어리그 팀은 상대에게 평균 12.07번의 슈팅을 허용했습니다.


## 토트넘의 상대 팀의 슈팅을 저지하는 능력은 프리미어리그 평균보다 살짝 높은 정도로 나타났습니다.

다른 여러 가지 수비에 관련된 지표가 있겠지만, 유의미한 분석 결과를 기대하기 어렵기에 더 이상 수비력에 관련된 분석은 하지 않겠습니다.


축구는 단순한 스포츠이기에, 골을 많이 넣고 골을 많이 막으면 이기는 건 당연합니다. 하지만 이런 득실차가 곧바로 순위 또는 경기의 승패를 결정짓는 것은 아닙니다. 득실차가 높아도 순위가 뒤쳐질 수 있고, 득실차가 낮아도 순위가 더 높을 수도 있죠. 이러한 것을 결정짓는 것은 결국 그 경기를 누가 이겼느냐입니다.

이번에는 이길 경기를 얼마나 이겼느냐, 즉 토트넘의 게임을 지배하는 능력을 분석해 보겠습니다.

전반전을 앞선 상태로 마친 경기 중, 승리로 끝마친 경기가 얼마나 되는지 살펴보겠습니다.

In [285]:
tot_rule_game = ((
    len(leage_tot_home[(leage_tot_home.HTR == 'H') & (leage_tot_home.FTR == 'H')])
    /
    len(leage_tot_home[(leage_tot_home.FTR == 'H')])
) + (
    len(leage_tot_away[(leage_tot_away.HTR == 'A') & (leage_tot_away.FTR == 'A')])
    /
    len(leage_tot_away[(leage_tot_away.FTR == 'A')])
)) / 2 * 100
print(f'토트넘이 전반전을 이긴 상태로 마친 경기를 승리로 마친 경기의 비율은 {tot_rule_game:.2f}% 입니다.')

winner_rule_game = ((
    len(leage_winner_20_21_home[(leage_winner_20_21_home.HTR == 'H') & (leage_winner_20_21_home.FTR == 'H')])
    /
    len(leage_winner_20_21_home[(leage_winner_20_21_home.FTR == 'H')])
) + (
    len(leage_winner_20_21_away[(leage_winner_20_21_away.HTR == 'A') & (leage_winner_20_21_away.FTR == 'A')])
    /
    len(leage_winner_20_21_away[(leage_winner_20_21_away.FTR == 'A')])
)) / 2 * 100

print(f'반면 프리미어리그 20-21시즌 우승팀의 같은 기록은 {winner_rule_game:.2f}% 였습니다.')

토트넘이 전반전을 이긴 상태로 마친 경기를 승리로 마친 경기의 비율은 56.90% 입니다.
반면 프리미어리그 20-21시즌 우승팀의 같은 기록은 70.88% 였습니다.
